vhdl/vgen_process_simulink_model.m

```diff
@@ -105,12 +113,39 @@
 %       with Quartus' embedded command shell instead?
 disp('Building kernel module.')
 cd(hdlpath)
-!make clean
-!make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf- 
-
+if ispc
+    if system('wsl.exe cd') == 0 
+        !wsl.exe make clean
+        !wsl.exe make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf-
+    else
+        disp("Windows Subsystem for Linux is currently required to automate building kernel modules")
+    end
+elseif isunix
+    !make clean
+    !make ARCH=arm CROSS_COMPILE=arm-linux-gnueabihf-
+else
+    disp('The current operating system is unsupported for automatically building kernel modules')
+end
 % TODO: this file now generates C code, but "vgen" make it seem like it is just VHDL still. This should be changed, and the repository should be reorganized a bit. 
 %       This file shouldn't live in the vhdl folder anymore.
 
+try
+    py.generate.generate_device_tree_overlay(hdlpath + "/quartus/" + mp.target_system + '_system.sopcinfo', mp.target_system + '.rbf')
+catch e
+    disp(getReport(e))
+end
+
+if ispc
+    if system('wsl.exe cd') == 0 
+        system("wsl.exe dtc -@ -O dtb -o " + mp.target_system + ".dtbo " + mp.target_system + ".dts");
+    else
+        disp("Windows Subsystem for Linux is currently required to automate compiling device tree overlays")
+    end
+elseif isunix
+    system("dtc -@ -O dtb -o " + mp.target_system + ".dtbo " + mp.target_system + ".dts");
```

Comment:
I think the device tree overlay should have the same name as the quartus project, so `mp.model_name`

Reply:
The sof file is named based on the project revision (by Quartus) and because I support both de-10 and audioblade based on project revisions the project revision isn't the project name. I could look at renaming project revisions to something like "echo_de10" to still support either target but make the rbf file have a more descriptive name than it currently does.